src/pybind/mgr/cephadm/serve.py

```diff
@@ -93,6 +93,8 @@ def serve(self) -> None:
 
                     self._check_daemons()
 
+                    self._check_certificates()
```

Comment:
Quick question here: is the `http` use case still handled? 
In OSP we can either deploy daemons using `http` (which means tls is terminated at haproxy level) or tls-everywhere, where tls applies also to the connection between `frontend` and `backend`.
I'm just worried we're not backward compatible with what we had in the past, and we need to introduce changes in the way haproxy is configured.

Reply:
> Quick question here: is the `http` use case still handled? In OSP we can either deploy daemons using `http` (which means tls is terminated at haproxy level) or tls-everywhere, where tls applies also to the connection between `frontend` and `backend`. I'm just worried we're not backward compatible with what we had in the past, and we need to introduce changes in the way haproxy is configured.

@fmount in the legacy code I'm afraid http was not option (unless I'm mistaken). cephadm will always try to use a cert/key when preparing grafana configuration.